plugin/pkg/shoot/quotavalidator/admission.go

```diff
@@ -307,11 +307,17 @@ func (q *QuotaValidator) determineAllocatedResources(quota garden.Quota, shoot g
 
 func (q *QuotaValidator) findShootsReferQuota(quota garden.Quota, shoot garden.Shoot) ([]garden.Shoot, error) {
 	var (
-		shootsReferQuota []garden.Shoot
-		secretBindings   []garden.SecretBinding
+		shootsReferQuota  []garden.Shoot
+		secretBindings    []garden.SecretBinding
+		allSecretBindings []*garden.SecretBinding
+		err               error
 	)
 
-	allSecretBindings, err := q.secretBindingLister.SecretBindings(v1.NamespaceAll).List(labels.Everything())
+	if quota.Spec.Scope == garden.QuotaScopeProject {
```

Comment:
Isn't it better to have a variable say `namespace` with default value `v1.NamespaceAll` and then 
```suggestion
	if quota.Spec.Scope == garden.QuotaScopeProject {
		namespace = shoot.Namespace
	}
	allSecretBindings, err := q.secretBindingLister.SecretBindings(namespace).List(labels.Everything())
```